### ISSUE TITLE
windows version?

### ISSUE THREAD

**Issue by antithing** [raulmur/ORB_SLAM#100]
Hi there, this looks amazing. Any chance that it would run on windows? Thanks!

**Comment by Beewe** [comment]
Yes, I've already have it running under windows. I'm planning to release this version on github, but first I have to merge the recent updates from this repo to mine and make sure everything is working. Also I've made it working as a library, so you can call TrackFrame(cv::Mat frame) and receive the camera pose.

**Comment by antithing** [comment]
Awesome! Thank you for all your work. :) I look forward to getting it
running.

**Comment by Beewe** [comment]
Hey, as you've maybe already seen there is a new version called ORB_SLAM2, which includes several new features, so I decided to make this Windows compatible. Check out my fork, I've got it already running: https://github.com/Beewe/ORB_SLAM2_Windows
Let me know if you still need/want a ORB_SLAM v1.

**Comment by antithing** [comment]
awesome. Thank you! I will check this out asap.

On Tue, Jan 26, 2016 at 12:58 PM, Benjamin <notifications@github.com> wrote:

> Hey, as you've maybe already seen there is a new version called ORB_SLAM2,
> which includes several new features, so I decided to make this Windows
> compatible. Check out my fork, I've got it already running:
> https://github.com/Beewe/ORB_SLAM2_Windows
> Let me know if you still need/want a ORB_SLAM v1.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/raulmur/ORB_SLAM/issues/100#issuecomment-174999824>.
>



-- 
Ben Sharp,

Vanishing Point Productions Ltd.

**Comment by antithing** [comment]
Hi, I cloned this dir and built the solution, but am getting errors on the
visual studio build..

Severity Code Description Project File Line Suppression State
Error C2280 'Eigen::Block<Eigen::Block<Eigen::Map<Eigen::Matrix<double,-1,-1,0,-1,-1>,0,Eigen::Stride<0,0>>,-1,-1,false,true>,-1,-1,false,true>
&Eigen::Block<Eigen::Block<Eigen::Map<Eigen::Matrix<double,-1,-1,0,-1,-1>,0,Eigen::Stride<0,0>>,-1,-1,false,true>,-1,-1,false,true>::operator
=(const
Eigen::Block<Eigen::Block<Eigen::Map<Eigen::Matrix<double,-1,-1,0,-1,-1>,0,Eigen::Stride<0,0>>,-1,-1,false,true>,-1,-1,false,true>
&)': attempting to reference a deleted function ORB_SLAM2
c:\libs\eigen\include\eigen\src\Core\SolveTriangular.h 194


From this section of Eigen:

if (copy) other = otherCopy;

Do you have any idea what that would be?

Thanks.


On Tue, Jan 26, 2016 at 1:51 PM, Ben Sharp <chemicalx001@gmail.com> wrote:

> awesome. Thank you! I will check this out asap.
>
> On Tue, Jan 26, 2016 at 12:58 PM, Benjamin <notifications@github.com>
> wrote:
>
>> Hey, as you've maybe already seen there is a new version called
>> ORB_SLAM2, which includes several new features, so I decided to make this
>> Windows compatible. Check out my fork, I've got it already running:
>> https://github.com/Beewe/ORB_SLAM2_Windows
>> Let me know if you still need/want a ORB_SLAM v1.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/raulmur/ORB_SLAM/issues/100#issuecomment-174999824>.
>>
>
>
>
> --
> Ben Sharp,
>
> Vanishing Point Productions Ltd.
>



-- 
Ben Sharp,

Vanishing Point Productions Ltd.

**Comment by Beewe** [comment]
Just having a guess: Maybe you use an outdated version of Eigen, see "Prerequisites" from the install instructions; you have to use Eigen at least in version 3.1.0. I just checked the linenumber 194 in version 3.0.7 and it has the same content as you described "if (copy) other = otherCopy;", whereas the linenumber 194 in version 3.2.4 (which I use) has a completly different content.

**Comment by antithing** [comment]
Got it, I just noticed that instruction as I saw your message. Thanks!

On Tue, Jan 26, 2016 at 2:57 PM, Benjamin <notifications@github.com> wrote:

> Just having a guess: Maybe you use an outdated version of Eigen, see
> "Prerequisites" from the install instructions; you have to use Eigen at
> least in version 3.1.0. I just checked the linenumber 194 in version 3.0.7
> and it has the same content as you described "if (copy) other =
> otherCopy;", whereas the linenumber 194 in version 3.2.4 (which I use) has
> a completly different content.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/raulmur/ORB_SLAM/issues/100#issuecomment-175058519>.
>



-- 
Ben Sharp,

Vanishing Point Productions Ltd.

**Comment by antithing** [comment]
Heya, sorry for all these questions! I have the kitti example building, but
it is crashing on loading the Vocabulary file. I get to:

'Loading ORB Vocabulary. This could take a while...' Then the program
crashes with 'abort has been called'. In a related question, if I run this
live using a camera input, do I need the Vocabulary file? If so, how do I
create this?

Thanks!

On Tue, Jan 26, 2016 at 3:11 PM, Ben Sharp <chemicalx001@gmail.com> wrote:

> Got it, I just noticed that instruction as I saw your message. Thanks!
>
> On Tue, Jan 26, 2016 at 2:57 PM, Benjamin <notifications@github.com>
> wrote:
>
>> Just having a guess: Maybe you use an outdated version of Eigen, see
>> "Prerequisites" from the install instructions; you have to use Eigen at
>> least in version 3.1.0. I just checked the linenumber 194 in version 3.0.7
>> and it has the same content as you described "if (copy) other =
>> otherCopy;", whereas the linenumber 194 in version 3.2.4 (which I use) has
>> a completly different content.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/raulmur/ORB_SLAM/issues/100#issuecomment-175058519>.
>>
>
>
>
> --
> Ben Sharp,
>
> Vanishing Point Productions Ltd.
>



-- 
Ben Sharp,

Vanishing Point Productions Ltd.

**Comment by Beewe** [comment]
You always need the Vocabulary file, it is used for place recognition and feature matching. There is already one in the Vocabulary directory which you you first have to extract because it's packed. Sorry I have never experienced such a crash yet. Did it crash immediately after "This could take a while..." or does the file load completly?

**Comment by antithing** [comment]
'This could take a while...'  When I debug, it actually errors on this line:

  F::fromString(m_nodes[nid].descriptor, ssd.str());

in TemplatedVoccabulary.h.


Thanks!


On Thu, Jan 28, 2016 at 12:50 PM, Benjamin <notifications@github.com> wrote:

> You always need the Vocabulary file, it is used for place recognition and
> feature matching. There is already one in the Vocabulary directory which
> you you first have to extract because it's packed. Sorry I have never
> experienced such a crash yet. Did it crash immediately after "This could
> take a while..." or does the file load completly?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/raulmur/ORB_SLAM/issues/100#issuecomment-176165673>.
>



-- 
Ben Sharp,

Vanishing Point Productions Ltd.

**Comment by antithing** [comment]
FIXED! I had some 64bit / 32bit dll conflicts. Rebuilding all sorted it.
Thanks again!

On Thu, Jan 28, 2016 at 2:18 PM, Ben Sharp <chemicalx001@gmail.com> wrote:

> 'This could take a while...'  When I debug, it actually errors on this
> line:
>
>   F::fromString(m_nodes[nid].descriptor, ssd.str());
>
> in TemplatedVoccabulary.h.
>
>
> Thanks!
>
>
> On Thu, Jan 28, 2016 at 12:50 PM, Benjamin <notifications@github.com>
> wrote:
>
>> You always need the Vocabulary file, it is used for place recognition and
>> feature matching. There is already one in the Vocabulary directory which
>> you you first have to extract because it's packed. Sorry I have never
>> experienced such a crash yet. Did it crash immediately after "This could
>> take a while..." or does the file load completly?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/raulmur/ORB_SLAM/issues/100#issuecomment-176165673>.
>>
>
>
>
> --
> Ben Sharp,
>
> Vanishing Point Productions Ltd.
>



-- 
Ben Sharp,

Vanishing Point Productions Ltd.
